FlagsmithClient/src/test/java/com/flagsmith/RealTimeUpdatesIntegrationTests.kt

```diff
@@ -0,0 +1,216 @@
+package com.flagsmith
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.content.res.Resources
+import android.graphics.Color
+import com.flagsmith.entities.FeatureStatePutBody
+import com.flagsmith.internal.FlagsmithEventTimeTracker
+import com.flagsmith.internal.FlagsmithRetrofitService
+import com.flagsmith.internal.FlagsmithRetrofitServiceTest
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+import java.io.File
+
+class RealTimeUpdatesIntegrationTests : FlagsmithEventTimeTracker {
+
+    private lateinit var flagsmith: Flagsmith
+
+    private lateinit var retrofitService: FlagsmithRetrofitServiceTest
+
+    // You'll need a valid account to test this
+    private val environmentKey = System.getenv("INTEGRATION_TESTS_ENVIRONMENT_KEY")
+        ?: throw Exception("INTEGRATION_TESTS_ENVIRONMENT_KEY not set")
+    private val apiToken = System.getenv("INTEGRATION_TESTS_API_TOKEN")
+        ?: throw Exception("INTEGRATION_TESTS_API_TOKEN not set")
+    private val authToken = "Token $apiToken"
+    private val featureId = System.getenv("INTEGRATION_TESTS_FEATURE_NAME")
+        ?: throw Exception("INTEGRATION_TESTS_FEATURE_NAME not set")
+    private val featureStateId = System.getenv("INTEGRATION_TESTS_FEATURE_STATE_ID")
+        ?: throw Exception("INTEGRATION_TESTS_FEATURE_STATE_ID not set")
+
+    @Mock
+    private lateinit var mockApplicationContext: Context
+
+    @Mock
+    private lateinit var mockContextResources: Resources
+
+    @Mock
+    private lateinit var mockSharedPreferences: SharedPreferences
+
+    // FlagsmithEventTimeTracker
+    override var lastFlagFetchTime: Double = 0.0
+
+    @Before
+    fun setup() {
+        setupMocks()
+
+        // We need the cache otherwise we'd be getting the new values from the server all the time
+        // Rather than seeing the realtime updates
```

Comment:
```suggestion
        // rather than seeing the realtime updates
```

Reply:
Also, does this mean that the realtime updates don't work without enabling the cache? I guess that makes sense, but we need to make sure it's documented.

Reply:
Hi @matthewelwell I've updated the comment in the integration test as it's more specific to that.

The real time updates should still come through without the cache, but if you tried to get a flag manually it would hit the API as the cache is on the HTTP client.